src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -33,13 +35,16 @@
     private final BackoffStrategy backoffStrategy;
     private final ObjectMapper objectMapper;
     private final CursorManager cursorManager;
+    private final NakadiReaderFactory nakadiReaderFactory;
 
     public NakadiClient(URI baseUri, ClientHttpRequestFactory clientHttpRequestFactory, BackoffStrategy backoffStrategy, ObjectMapper objectMapper, CursorManager cursorManager) {
         this.baseUri = baseUri;
         this.clientHttpRequestFactory = clientHttpRequestFactory;
         this.backoffStrategy = backoffStrategy;
```

Comment:
`BackoffStrategy` can maybe be moved to the factory, I don't think it's used outside of `NakadiReader`